uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -1744,26 +1745,51 @@ public function testLeaveRequestCanNotBeCreatedWhenLeaveRequestDateIsGreaterThan
       'period_id' => $period->id
     ]);
 
-    $this->createLeaveBalanceChange($periodEntitlement->id, 4);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 30);
+    $periodStartDate1 = date('2016-06-01');
+    $periodEndDate1 = date('2016-06-08');
+
+    $periodStartDate2 = date('2016-06-09');
+    $periodEndDate2 = date('2016-06-12');
+
+    $periodStartDate3 = date('2016-06-14');
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      [
+        'period_start_date' => $periodStartDate1,
+        'period_end_date' => $periodEndDate1
+      ]
+    );
 
     HRJobContractFabricator::fabricate(
       ['contact_id' => $periodEntitlement->contact_id],
       [
-        'period_start_date' => '2016-01-01',
-        'period_end_date' => '2016-06-30'
+        'period_start_date' => $periodStartDate2,
+        'period_end_date' => $periodEndDate2
       ]
     );
 
-    WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => 1]);
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      ['period_start_date' => $periodStartDate3]
+    );
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $periodEntitlement->contact_id,
+      'pattern_id' => $workPattern->id
+    ]);
 
-    //The to_date of the leave request is outside the contract dates
+    //The from date and to date overlaps the three job contracts with a lapse of
+    // 1 day without any contract between the last two contracts
     LeaveRequest::create([
       'type_id' => $periodEntitlement->type_id,
       'contact_id' => $periodEntitlement->contact_id,
       'status_id' => 1,
-      'from_date' => CRM_Utils_Date::processDate('2016-06-29'),
+      'from_date' => CRM_Utils_Date::processDate('2016-06-07'),
       'from_date_type' => $this->leaveRequestDayTypes['all_day']['value'],
-      'to_date' => CRM_Utils_Date::processDate('2016-07-01'),
+      'to_date' => CRM_Utils_Date::processDate('2016-06-16'),
       'to_date_type' => $this->leaveRequestDayTypes['all_day']['value'],
       'request_type' => LeaveRequest::REQUEST_TYPE_LEAVE
     ]);
```

Comment:
I wonder if we should also add new test to ensure we can create a leave request overlapping two contracts as long as there's no lapse between them